src/main/java/org/opensearch/security/securityconf/ConfigModelV6.java

```diff
@@ -542,6 +542,26 @@ public boolean impliesTypePermGlobal(
             roles.stream().forEach(p -> ipatterns.addAll(p.getIpatterns()));
             return ConfigModelV6.impliesTypePerm(ipatterns, resolved, user, actions, resolver, cs);
         }
+
+        @Override
+        public boolean isPermittedOnSystemIndex(String indexName) {
+            boolean isPatternMatched = false;
+            boolean isPermitted = false;
+            for (SecurityRole role : roles) {
+                for (IndexPattern ip : role.getIpatterns()) {
+                    WildcardMatcher wildcardMatcher = WildcardMatcher.from(ip.indexPattern);
+                    if (wildcardMatcher.test(indexName)) {
+                        isPatternMatched = true;
```

Comment:
This would match against `*` pattern. Correct me if i'm wrong. we do not want that. We want to explicitly match against an system index pattern. For example, `*` should not be matched but `.system_index*` should be matched.

Reply:
It seems that `*` should matched `.system_index1234`. I checked documents and codes.

https://opensearch.org/docs/latest/security/access-control/permissions/#system-index-permissions

> The following example shows a role that grants access to all system indexes:
```
index_permissions:
    - index_patterns:
        - '*'
    - allowed_actions:
        - 'system:admin/system_index'
```